src/adapter/src/client.rs

```diff
@@ -215,6 +216,31 @@ impl Client {
         &self.metrics
     }
 
+    pub async fn append_webhook(
+        &self,
+        database: String,
+        schema: String,
+        name: String,
+    ) -> Result<Option<(CollectionAppender, ColumnType, Option<ColumnType>)>, AdapterError> {
+        let (tx, rx) = oneshot::channel();
+
+        // Send our request.
+        self.send(Command::AppendWebhook {
+            database,
+            schema,
+            name,
+            tx,
+        });
+
+        // Using our one shot channel to get the result.
+        let result = rx
+            .await
+            .map_err(|_| AdapterError::Internal("failed to receive webhook response".to_string()))??
```

Comment:
We're only mapping to an internal error if receiving on the channel fails, other errors like a typo in the webhook source, are passed along. It's hard to see this though because it's done with the double `?`. I reformatted this a bit, hopefully that helps